subsys/bluetooth/host/scan.c

```diff
@@ -507,50 +551,157 @@ static uint8_t get_adv_type(uint8_t evt_type)
 	}
 }
 
+/* Convert extended adv report evt_type field to adv props */
+static uint16_t get_adv_props_extended(uint16_t evt_type)
+{
+	/* Converts from BT_HCI_LE_ADV_EVT_TYPE_* to BT_GAP_ADV_PROP_*
+	 * The first 4 bits are the same (conn, scan, direct, scan_rsp).
+	 * Bit 4 must be flipped as the meaning of 1 is opposite (legacy -> extended)
+	 * The rest of the bits are zeroed out.
+	 */
+	return (evt_type ^ BT_HCI_LE_ADV_EVT_TYPE_LEGACY) & BIT_MASK(5);
+}
+
+static void create_ext_adv_info(struct bt_hci_evt_le_ext_advertising_info const *const evt,
+				struct bt_le_scan_recv_info *const scan_info)
+{
+	scan_info->primary_phy = bt_get_phy(evt->prim_phy);
+	scan_info->secondary_phy = bt_get_phy(evt->sec_phy);
+	scan_info->tx_power = evt->tx_power;
+	scan_info->rssi = evt->rssi;
+	scan_info->sid = evt->sid;
+	scan_info->interval = sys_le16_to_cpu(evt->interval);
+	scan_info->adv_type = get_adv_type(evt->evt_type);
+	scan_info->adv_props = get_adv_props_extended(evt->evt_type);
+}
+
 void bt_hci_le_adv_ext_report(struct net_buf *buf)
 {
 	uint8_t num_reports = net_buf_pull_u8(buf);
 
-	BT_DBG("Adv number of reports %u",  num_reports);
+	BT_DBG("Adv number of reports %u", num_reports);
 
 	while (num_reports--) {
 		struct bt_hci_evt_le_ext_advertising_info *evt;
-		struct bt_le_scan_recv_info adv_info;
+		struct bt_le_scan_recv_info scan_info;
+		uint8_t size_to_copy;
+		bool truncate;
+		bool legacy_pdu_used;
+		uint16_t data_status;
+		bool is_report_complete;
+		bool more_to_come;
+		bool ctrl_truncated;
 
 		if (buf->len < sizeof(*evt)) {
 			BT_ERR("Unexpected end of buffer");
 			break;
 		}
 
 		evt = net_buf_pull_mem(buf, sizeof(*evt));
+		legacy_pdu_used = evt->evt_type & BT_HCI_LE_ADV_EVT_TYPE_LEGACY;
+		data_status = BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS(evt->evt_type);
+		is_report_complete = data_status == BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_COMPLETE;
+		more_to_come = data_status == BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_PARTIAL;
+		ctrl_truncated = data_status == BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_INCOMPLETE;
+
+		if (legacy_pdu_used) {
+			/* Complete advertising report.
+			 * Does not need reassembly, create event immediately
+			 */
+			create_ext_adv_info(evt, &scan_info);
+			le_adv_recv(&evt->addr, &scan_info, buf, evt->length);
+			continue;
+		}
+
+		const bool is_new_advertiser =
+			!p_reassembling_advertiser ||
+			!fragmented_advertisers_equal(p_reassembling_advertiser, &evt->addr,
+						      evt->sid);
```

Comment:
I am personally in favor of declaring things as late as possible, but it seems the rest of the codebase declares everything on top. Simplest change here is to move this one up.

Reply:
There isn't a specific rule for that, and since there isn't then the rule is to follow the file's code style (which seems to be declarations at top) :)